onnxruntime/core/platform/posix/env.cc

```diff
@@ -165,24 +183,16 @@ class PosixThread : public EnvThread {
           ORT_THROW("pthread_attr_setstacksize failed, error code: ", err_no, " error msg: ", err_msg);
         }
       }
-      s = pthread_create(&hThread, &attr, ThreadMain,
-                         new Param{name_prefix, index, start_address, param, thread_options});
+      if (!thread_options.affinity.empty() && gsl::narrow<size_t>(index) < thread_options.affinity.size()) {
+        param_ptr->affinity_mask = thread_options.affinity[index];
+      }
+
+      s = pthread_create(&hThread, &attr, ThreadMain, param_ptr.get());
       if (s != 0) {
         auto [err_no, err_msg] = GetSystemError();
         ORT_THROW("pthread_create failed, error code: ", err_no, " error msg: ", err_msg);
       }
-#if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX)
```

Comment:
It's not very clear why this block was deleted from here.

Reply:
The block was removed from here, because we can not set affinity before the thread is created. And we cannot throw after the thread is created, because without an instance owning the thread it becomes orphaned and we cannot join any longer making the process hang. Moving affinity setting to the actual thread gives us a little more flexibility, although if we do not throw, we can still have it in the __Ctor